include/odp/api/spec/classification.h

```diff
@@ -780,7 +780,7 @@ int odp_cos_queue_set(odp_cos_t cos, odp_queue_t queue);
 * @param cos           CoS handle
 *
 * @retval Queue handle associated with the given class-of-service
-* @retval ODP_QUEUE_INVALID on failure
+* @retval ODP_QUEUE_INVALID on failure, or if the queue has not been set
```

Comment:
I think we may need to further clarify when CoS may not have queue set. Obviously, a ODP_COS_ACTION_DROP CoS will not have destination queue (API says that queue handle is ignored in CoS params).

However, I'm not sure if ENQUEUE action can have destination queue not set. CoS params expect that it is set:

> Mapping used when num_queue = 1, hashing is disabled in
> this case and application has to configure this queue and
> packets are delivered to this queue
> 
> odp_queue_t queue;

Reply:
Added some clarifications in v3 (last commit), not sure if it's what you had in mind.

Reply:
That addition looks good.